### ISSUE TITLE
Batch norm

### ISSUE THREAD

**Issue by Joker316701882** [cameronfabbri/Improved-Wasserstein-GAN#1]
Hi dear author!

I notice that in your code, there is no update_ops for updating batch norm statistics, and update collections for batch norm is also not set as None. Besides, should is_training be set as a placeholder to correctly use batch norm?

Did I miss something?

**Comment by IPNUISTlegal** [comment]
@Joker316701882 
I also notice that batch norm should  be used in training mode .It seems he pays no attention to this issue.
Do you have idea to solve it?
sry, i am noob in python

**Comment by cameronfabbri** [comment]
I do use batch norm in the generator, but because of the penalized gradient with respect to the input for the discriminator, no batch norm is used. The authors explain at the end of page 3 https://arxiv.org/pdf/1704.00028.pdf

**Comment by IPNUISTlegal** [comment]
I think you may have misunderstood my issue. tf.contrib.layers.batch_norm has one parameter is_training. https://www.tensorflow.org/api_docs/python/tf/contrib/layers/batch_norm
Batch Normalization should used in training process rather than test processs. I am sorry that  Batch Normalization distinguish between training set or test set. In your code  `   enc_conv1 = tcl.batch_norm(enc_conv1)` , it's weird without is_training  set.

**Comment by cameronfabbri** [comment]
Ah I see. Without it set it defaults to true (as per your link). With generative models you want to use batch norm in both training and testing. You can easily try it out to see if the results improve when not using it during testing, but I don't think they will.
